### PR TITLE
Renamed account properties to account restrictions

### LINKED ISSUE #49: rename account_properties to restriction_account to match server and catbuffer

### PR DESCRIPTION
**Pull request by martinayora**
Fixes #49

### CODE REVIEW

File: catapult-sdk/src/plugins/accountRestrictions.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+/** @module plugins/accountRestrictions */
+const EntityType = require('../model/EntityType');
+const ModelType = require('../model/ModelType');
+const sizes = require('../modelBinary/sizes');
+
+const constants = { sizes };
+
+const accountRestrictionTypeBlockOffset = 128;
+const accountRestrictionTypeFlags = Object.freeze({
```
Comment (Jaguar0625) [inline catapult-sdk/src/plugins/accountRestrictions.js]: did you intentionally change to lower case ?

File: catapult-sdk/src/plugins/accountRestrictions.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+const accountRestrictionsCreateBaseCodec = valueCodec => ({
+	deserialize: parser => {
+		const transaction = {};
+		transaction.restrictionType = parser.uint8();
+		transaction.modifications = [];
+		const restrictionsCount = parser.uint8();
+		for (let i = 0; i < restrictionsCount; ++i) {
+			transaction.modifications.push({
+				type: parser.uint8(),
```
Comment (Jaguar0625) [inline catapult-sdk/src/plugins/accountRestrictions.js]: why did you change this from modificationType?

File: catapult-sdk/src/plugins/accountRestrictions.js (hunk 1 above)
Comment (martinayora) [reply catapult-sdk/src/plugins/accountRestrictions.js]: no, I have changed it right now.

File: catapult-sdk/src/plugins/accountRestrictions.js (hunk 2 above)
Comment (martinayora) [reply catapult-sdk/src/plugins/accountRestrictions.js]: I have change it to "type" only because this is how it is in mongodb schema now.

Comment (Vektrat) [reply catapult-sdk/src/plugins/accountRestrictions.js]: PR looks fine except for this doubt, please confirm ^^

Comment (martinayora) [reply catapult-sdk/src/plugins/accountRestrictions.js]: I see you changed to restrictionType in mongodb collection. I have updated.
